http/router.go

```diff
@@ -129,14 +129,17 @@ func panicMW(api *kithttp.API) func(http.Handler) http.Handler {
 	}
 }
 
-var panicLogger *zap.Logger
+var panicLogger = zap.NewNop()
 var panicLoggerOnce sync.Once
 
 // getPanicLogger returns a logger for panicHandler.
 func getPanicLogger() *zap.Logger {
 	panicLoggerOnce.Do(func() {
```

Comment:
Should this function be extracted somewhere?

Reply:
Looked into it. These files are nearly identical and I think we could consolidate into `kit/transport/http`, but would prefer to leave it for a separate issue. Made  #20712 to track it